src/composables/useTransactions.ts

```diff
@@ -0,0 +1,327 @@
+import { computed, ref } from 'vue';
+import { merge, orderBy } from 'lodash';
+import { TransactionReceipt } from '@ethersproject/providers';
+import { useI18n } from 'vue-i18n';
+
+import { configService } from '@/services/config/config.service';
+import { gnosisOperator } from '@/services/gnosis/operator.service';
+import useVueWeb3 from '@/services/web3/useVueWeb3';
+import { OrderMetaData } from '@/services/gnosis/types';
+import { gnosisExplorer } from '@/services/gnosis/explorer.service';
+
+import { lsGet, lsSet } from '@/lib/utils';
+
+import LS_KEYS from '@/constants/local-storage.keys';
+import useNotifications from './useNotifications';
+
+const DAY_MS = 86_400_000;
+
+export type TransactionStatus =
+  | 'pending'
+  | 'confirmed'
+  | 'expired'
+  | 'cancelling'
+  | 'cancelled';
+
+export type TransactionAction =
+  | 'claim'
+  | 'approve'
+  | 'trade'
+  | 'invest'
+  | 'withdraw';
+
+export type TransactionType = 'order' | 'tx';
+
+export type TxReceipt = Pick<
+  TransactionReceipt,
+  | 'blockHash'
+  | 'blockNumber'
+  | 'contractAddress'
+  | 'from'
+  | 'status'
+  | 'to'
+  | 'transactionHash'
+  | 'transactionIndex'
+>;
+
+export type Transaction = {
+  id: string;
+  action: TransactionAction;
+  type: TransactionType;
+  receipt?: OrderMetaData | TxReceipt;
+  summary: string;
+  addedTime: number;
+  confirmedTime?: number;
+  from: string;
+  lastCheckedBlockNumber?: number;
+  status: TransactionStatus;
+};
+
+export type NewTransaction = Pick<
+  Transaction,
+  'id' | 'type' | 'summary' | 'receipt' | 'action'
+>;
+
+const networkId = Number(configService.env.NETWORK);
+
+export type TransactionsMap = Record<string, Transaction>;
+
+export type TransactionState = {
+  [networkId: number]: TransactionsMap;
+};
+
+const PERSIST_TRANSACTIONS = false;
```

Comment:
@garethfuller I'm disabling it for now until we have all the "structure" correct and reviewed... I wouldn't want to persist a `summary` prop if we are going to change it soon.